### PR TITLE
Reapply changes made to XmlSerializer

### LINKED ISSUE #11768: Reapply changes made to Xml libraries since the move to System.Private.Xml since they got lost in merge from dev/api
When we moved the implementation of Xml from the old contracts into System.Private.Xml, we lost the history since to git's point of view, they are two different files now. The problem is that this move happened in a different branch (dev/xml) so when we merged that back to dev/api and then to master, we lost all of the changes that were made in master after the move. This work item is to track the work that must be done to re-apply these changes back to the files now under System.Private.Xml.

cc: @weshaggard @danmosemsft @krwq @shmao

### PR DESCRIPTION
**Pull request by shmao**
Reapply changes made to XmlSerializer since the move to System.Private.Xml since they got lost in merge from dev/api.

Fix #11768.

/cc: @mconnew @huanwu @zhenlan @joperezr

### CODE REVIEW

File: src/System.Private.Xml/src/System/Xml/Serialization/ReflectionXmlSerializationReader.cs [hunk 1]
```diff
@@ -726,9 +726,9 @@ private object WriteEnumMethod(EnumMapping mapping, Func<string> readFunc)
             }
         }
 
-        private Dictionary<string, long> WriteHashtable(EnumMapping mapping, string name)
+        private Hashtable WriteHashtable(EnumMapping mapping, string name)
```
Comment (justinvp) [inline src/System.Private.Xml/src/System/Xml/Serialization/ReflectionXmlSerializationReader.cs]: Why Hashtable?

File: src/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs [hunk 2]
```diff
@@ -258,6 +279,16 @@ public XmlSerializer(Type type, string defaultNamespace)
             {
                 this.innerSerializer = contract.GetSerializer(type);
             }
+			else if (ReflectionMethodEnabled)
```
Comment (justinvp) [inline src/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs]: Indentation seems off

File: src/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs [hunk 3]
```diff
@@ -614,7 +718,19 @@ public static XmlSerializer[] FromMappings(XmlMapping[] mappings)
...
+            for(int i=0;i<mappings.Length;i++)
+            {
+                serializers[i] = new XmlSerializer();
+                serializers[i].rootType = type;
+                serializers[i]._mapping = mappings[i];
+            }
+
+            return serializers;
+#else
+			XmlSerializerImplementation contract = null;
```
Comment (justinvp) [inline src/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs]: Indentation seems off

File: src/System.Private.Xml/src/System/Xml/Serialization/ReflectionXmlSerializationReader.cs (hunk 1 above)
Comment (shmao) [inline src/System.Private.Xml/src/System/Xml/Serialization/ReflectionXmlSerializationReader.cs]: XmlSerializationReader.ToEnum() used above uses HashTable.

**Comment by joperezr** [comment]
Thanks a lot for doing this @shmao! I am curious though, did you re-write the added code for XmlSerializer or did you just diffed what was removed after the merge and then added that back? I ask because I'll re-apply the rest of the changes that got lost for this assembly and I would like to understand how was your process.

**Comment by shmao** [comment]
@joperezr I tried diff'ing the serialization folder in System.Private.Xml with the old System.Xml.XmlSerializer source code. But I got a lot of differences since the code in System.Private.Xml was from desktop with more available APIs. So I ended up with checking the history of release\1.1.0 branch and found the commits made after forking master and dev/api and re-wrote those changes.

**Comment by joperezr** [comment]
I see, sounds good, Thanks again for doing this and I'll be re-adding back the rest of the changes that were lost soon.

**Pull request merged**